onnxruntime/core/providers/js/operators/range.cc

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/js/js_kernel.h"
+#include "range.h"
+
+namespace onnxruntime {
+namespace js {
+ONNX_OPERATOR_KERNEL_EX(
+    Range,
+    kOnnxDomain,
+    11,
+    kJsExecutionProvider,
+    KernelDefBuilder()
+        .TypeConstraint("T", DataTypeImpl::GetTensorType<int32_t>()),
```

Comment:
Can we specify that the inputs memory type is cpu `InputMemoryType(OrtMemTypeCPU, xxx)`? So in the js side, we don't need to call `downloadSync` explicitly?

Reply:
I think `InputMemoryType(OrtMemTypeCPU, xxx)` means it is input tensor whose data indeed on the cpu side, and need to upload to GPU,  likes the weights data of conv operators. But in this PR, the tensors of `start`, `limit` and `delta` may be the output of one node and the input of the latter node, these tensors' data is on GPU side.

Reply:
@fs-eire Does `InputMemoryType(OrtMemTypeCPU, xxx)` support download data from GPU if the tensor's data is in GPU?

I came up with an example of the opposite. If those inputs' data are all in cpu, in current implementation, it will become upload data to gpu, then download them back, right? So maybe we need to check the Segment Anything model to see which case it uses.

Reply:
I think you are right, if the data is actually on the cpu side, the flow is still uploading the data to gpu side firstly, and then download the data from gpu to cpu, it indeed loses performance here. @fs-eire, any comments here?

Reply:
`InputMemoryType(OrtMemTypeCPU, xxx)` will ensure the data is on CPU.

If it is originally on CPU, then no operation will be done.
If it is originally on GPU, ORT will launch the data transfer to copy data from GPU to CPU (which calls into `downloadAsync()` )